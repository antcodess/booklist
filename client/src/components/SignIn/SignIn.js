import React, { useState } from 'react';
import { connect } from 'react-redux';

import FormInput from '../FormInput/FormInput';
import CustomButton from '../CustomButton/CustomButton';

import { googleSignInStart, emailSignInStart } from '../../redux/user/user.actions'

import './SignIn.scss';

export const SignIn = ({ googleSignInStart, emailSignInStart }) => {
  const [userCredentials, setUserCredentials] = useState({ email: '', password: '' });

  const { email, password } = userCredentials;

  const onInputSubmit = (event) => {
    event.preventDefault();
    emailSignInStart(email, password)
  };

  const onInputChange = (event) => {
    const { name, value } = event.target;
    setUserCredentials({...userCredentials, [name]: value })
  };

  return (
    <section className='sign-in'>
      <h2>SignIn</h2>
      <h3>Already have an account?</h3>
      <span>Sign in with your email and password.</span>

      <form onSubmit={onInputSubmit}>
        <FormInput 
          name='email'
          type='email'
          value={email}
          onChange={onInputChange} 
          label='Email'
          placeholder='Email' 
          required 
        />
        <FormInput 
          name='password'
          type='password'
          value={password}
          onChange={onInputChange} 
          label='Password'
          placeholder='Password' 
          required 
        />
        <div className='buttons'>
          <CustomButton 
          type='submit'
          isRegularSignIn>Sign In</CustomButton>
          <CustomButton 
          type='button'
          onClick={googleSignInStart}
          isGoogleSignIn>Sign In With Google</CustomButton>
        </div>
      </form>
    </section>
  );
};

export const mapDispatchToProps = dispatch => ({
  googleSignInStart: () => dispatch(googleSignInStart()),
  emailSignInStart: (email, password) => dispatch(emailSignInStart({ email, password }))
});

export default connect(null, mapDispatchToProps)(SignIn);