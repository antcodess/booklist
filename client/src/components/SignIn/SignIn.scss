.sign-in {
  width: 48%;
  display: flex;
  flex-direction: column;

  h3 {
    margin: 20px 0 10px;
    text-align: left;
    font-size: 1.2rem;
  }

  span {
    text-align: left;
    font-size: 0.90rem;
  }

  .buttons {
    display: flex;
    justify-content: space-between;
  }
}

@media only screen and (max-width: 882px) {
  .buttons {
    flex-direction: column;
  }
}

@media only screen and (max-width: 720px) {
  .sign-in {
    width: 100%;
    margin-bottom: 40px;
  }
}