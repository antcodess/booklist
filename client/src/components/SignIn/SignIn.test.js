import React from 'react';
import { shallow } from 'enzyme';
import { SignIn } from './SignIn';
import { mapDispatchToProps } from './SignIn';

describe('SignIn Component', () => {
  let wrapper;
  let mockGoogleSignInStart;
  let mockEmailSignInStart;
  let dummyFunction = jest.fn();

  const mockPayload = {
    type: "EMAIL_SIGN_IN_START",
    payload: {
      email: 'email@example.com', 
      password: 'password'
    }
  };
  
  beforeEach(() => {

    mockGoogleSignInStart = jest.fn();
    mockEmailSignInStart = jest.fn();

    const mockProps = {
      googleSignInStart: mockGoogleSignInStart, 
      emailSignInStart: mockEmailSignInStart,
      userCredentials: {
        email: 'email@example.com', 
        password: 'password'
      }
    };
    
    wrapper = shallow(<SignIn { ...mockProps } />);
  });

  it('should render SignIn Component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should dispatch googleSignInStart action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).googleSignInStart();
    expect(dispatch.mock.calls[0][0]).toEqual({type: "SIGN_IN_START"});
  });

  it('should dispatch emailSignInStart action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).emailSignInStart('email@example.com', 'password');
    expect(dispatch.mock.calls[0][0]).toEqual(mockPayload);
  });
});
