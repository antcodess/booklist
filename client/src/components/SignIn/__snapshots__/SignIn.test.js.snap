// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Component should render SignIn Component 1`] = `ShallowWrapper {}`;
